subsys/bluetooth/controller/ll_sw/ull_iso_internal.h

```diff
@@ -4,19 +4,31 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
 /* Calculate ISO PDU buffers required considering SDU fragmentation */
-/* FIXME: Calculation considering both Connected and Broadcast ISO PDU
- *        fragmentation.
- */
-#if defined(CONFIG_BT_CTLR_CONN_ISO)
-#define BT_CTLR_ISO_TX_BUFFERS (((CONFIG_BT_CTLR_CONN_ISO_SDU_LEN_MAX + \
-				  CONFIG_BT_CTLR_CONN_ISO_PDU_LEN_MAX - 1U) / \
-				 CONFIG_BT_CTLR_CONN_ISO_PDU_LEN_MAX) * \
+#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+#define BT_CTLR_ISO_SDU_LEN_MAX ((CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE * \
+				  CONFIG_BT_CTLR_ISO_TX_BUFFERS) - \
+				 BT_HCI_ISO_SDU_HDR_SIZE)
+#if defined(CONFIG_BT_CTLR_ADV_ISO) && defined(CONFIG_BT_CTLR_CONN_ISO)
+#define BT_CTLR_ISO_TX_BUFFERS (DIV_ROUND_UP(BT_CTLR_ISO_SDU_LEN_MAX, \
+					     MIN((CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE - \
+						  BT_HCI_ISO_SDU_TS_HDR_SIZE), \
+						 MIN(CONFIG_BT_CTLR_ADV_ISO_PDU_LEN_MAX, \
+						     CONFIG_BT_CTLR_CONN_ISO_PDU_LEN_MAX))) * \
+				CONFIG_BT_CTLR_ISO_TX_BUFFERS)
+#elif defined(CONFIG_BT_CTLR_ADV_ISO)
+#define BT_CTLR_ISO_TX_BUFFERS (DIV_ROUND_UP(BT_CTLR_ISO_SDU_LEN_MAX, \
+					     MIN((CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE - \
+						  BT_HCI_ISO_SDU_TS_HDR_SIZE), \
+						 CONFIG_BT_CTLR_ADV_ISO_PDU_LEN_MAX)) * \
+				CONFIG_BT_CTLR_ISO_TX_BUFFERS)
+#else /* CONFIG_BT_CTLR_CONN_ISO */
+#define BT_CTLR_ISO_TX_BUFFERS (DIV_ROUND_UP(BT_CTLR_ISO_SDU_LEN_MAX, \
+					     MIN((CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE - \
+						  BT_HCI_ISO_SDU_TS_HDR_SIZE), \
+						 CONFIG_BT_CTLR_CONN_ISO_PDU_LEN_MAX)) * \
 				CONFIG_BT_CTLR_ISO_TX_BUFFERS)
-#else /* !CONFIG_BT_CTLR_CONN_ISO */
-#define BT_CTLR_ISO_TX_BUFFERS CONFIG_BT_CTLR_ISO_TX_BUFFERS
-#endif /* !CONFIG_BT_CTLR_CONN_ISO */
+#endif /* CONFIG_BT_CTLR_CONN_ISO */
```

Comment:
There's gotta be better way than this :D 

I don't have one on hand right now, but still...

Reply:
agree. And this is not feasible either, the RAM usage is too high (see CI failures) and need to limit the calculated tx buffer counts.